src/Stratis.Bitcoin.Features.BlockStore/AddressIndexing/AddressIndexerCV.cs

```diff
@@ -0,0 +1,103 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using NBitcoin;
+using Stratis.Bitcoin.Builder.Feature;
+using Stratis.Bitcoin.Controllers.Models;
+using Stratis.Bitcoin.Features.BlockStore.Models;
+using Stratis.Bitcoin.Features.Consensus.CoinViews;
+using Stratis.Bitcoin.Interfaces;
+
+namespace Stratis.Bitcoin.Features.BlockStore.AddressIndexing
+{
+    public class AddressIndexerCV : IAddressIndexer
+    {
+        private readonly Network network;
+        private readonly ICoinView coinView;
+        private readonly ChainIndexer chainIndexer;
+        private readonly IScriptAddressReader scriptAddressReader;
+
+        public ChainedHeader IndexerTip => GetTip();
+
+        public IFullNodeFeature InitializingFeature { set; private get; }
+
+        public AddressIndexerCV(Network network, ChainIndexer chainIndexer, IScriptAddressReader scriptAddressReader, ICoinView coinView)
+        {
+            this.network = network;
+            this.coinView = coinView;
+            this.chainIndexer = chainIndexer;
+            this.scriptAddressReader = scriptAddressReader;
+        }
+
+        private ChainedHeader GetTip()
+        {
+            this.coinView.Sync(this.chainIndexer);
+
+            return this.chainIndexer[this.coinView.GetTipHash().Hash];
+        }
+
+        public void Initialize()
+        {
+        }
+
+        private TxDestination AddressToDestination(string address)
+        {
+            var bitcoinAddress = BitcoinAddress.Create(address, this.network);
+            return this.scriptAddressReader.GetDestinationFromScriptPubKey(this.network, bitcoinAddress.ScriptPubKey).Single();
+        }
+
+        public AddressBalancesResult GetAddressBalances(string[] addresses, int minConfirmations = 0)
+        {
+            return new AddressBalancesResult()
+            {
+                Balances = addresses.Select(address => new AddressBalanceResult()
+                {
+                    Address = address,
+                    Balance = new Money(this.coinView.GetBalance(AddressToDestination(address)).First().satoshis)
```

Comment:
I guess it doesn't make much difference for current use cases, but what if `AddressToDestination` returns null for scriptPubKey formats that don't fall into one of the recognised categories?

Reply:
Fixed.